examples/reddit_pagerank.cpp

```diff
@@ -0,0 +1,267 @@
+// Copyright 2019-2021 Lawrence Livermore National Security, LLC and other YGM
+// Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: MIT
+
+#include <fstream>
+#include <iostream>
+#include <iomanip>
+
+#include <math.h>
+#include <ygm/comm.hpp>
+
+#include <ygm/container/map.hpp>
+#include <ygm/io/ndjson_parser.hpp>
+#include <ygm/container/experimental/maptrix.hpp>
+
+#include <ygm/utility.hpp>
+
+double compute_norm(
+  ygm::container::map<boost::json::string, double> &pr_old, 
+  ygm::container::map<boost::json::string, double> &pr_new,
+  ygm::comm &world) {
+
+  using map_type = ygm::container::map<boost::json::string, double>;
+
+  int N_old = pr_old.size();
+  int N_new = pr_new.size();
+  //std::cout << "Prev iter size: " << N_old << ", Current iter size: " << N_new << std::endl;
+
+  auto pr_old_ptr = pr_old.get_ygm_ptr();
+  auto pr_new_ptr = pr_new.get_ygm_ptr();
+
+  map_type norm_map(world);
+  auto norm_map_ptr = norm_map.get_ygm_ptr();
```

Comment:
You can do this without a norm_map that gathers residuals component by component. Instead, define a `static double` that holds the piece of the squared norm local to the rank.

A static variable can be used inside of a lambda (even one spawned by a visit) without being explicitly passed (either as an argument or by capturing).

Replace the definition of norm_map with
`static double local_norm_squared;
local_norm_squared = 0.0;`

Reply:
Changed this, thanks!